app/models/vm.rb

```diff
@@ -121,6 +121,19 @@ def supported_consoles
     }
   end
 
+  def validate_v2v_migration
+    # VM has not been migrated before
+    return TransformationMapping::VM_VALID if ServiceResource.where(:resource => self).none? { |rsc| rsc.service_template.kind_of?(ServiceTemplateTransformationPlan) }
```

Comment:
Consider to call `ServiceResource.where(...)` only once, and analyze the status for complete, failed, or other with code.